sale_delivery_date/models/sale_order_line.py

```diff
@@ -17,13 +17,53 @@ class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
     def _prepare_procurement_values(self, group_id=False):
+        # Update 'date_planned' and 'date_deadline' with respect to:
+        #   - the warehouse or partner cutoff time
+        #   - the warehouse calendar
+        #   - the delivery time window of the customer
+        # knowing that 'date_planned' is when the transfer starts to be processed
+        # internally and 'date_deadline' is when the goods will be delivered to
+        # the customer.
+        # There's 2 cases here:
+        #   1) commitment_date is set, compute date_planned from date_deadline
+        #   2) commitment_date isn't set, compute date_planned and date_deadline
         res = super()._prepare_procurement_values(group_id=group_id)
+        if self.order_id.commitment_date:
+            res = self._prepare_procurement_values_commitment_date(res)
+        else:
+            res = self._prepare_procurement_values_no_commitment_date(res)
+        return res
+
+    def _prepare_procurement_values_commitment_date(self, res):
+        """Set 'date_planned' according to 'commitment_date'.
+
+        We want to find the date when we could start working on the transfers
+        so we will be able to ship the goods to the customer in respect to the
+        `commitment_date`.
+        The 'date_deadline' is already set to the 'commitment_date' (that should
+        already fit the partner's time window if any).
+        """
+        # TODO
+        # 1) commitment_date - security_lead = order ready to be shipped {1}
+        # 2) {1} - workload
+        # 3) while {2} isn't a working day, remove 1 day
+        # 4) apply cutoff (with `keep_same_day` param)
+        return res
+
+    def _prepare_procurement_values_no_commitment_date(self, res):
+        """Set 'date_planned' and 'date_deadline' if no 'commitment_date'."""
+        res["date_planned"] = (
+            self.order_id.date_order
+            + timedelta(days=self.customer_lead or 0.0)
+            - timedelta(days=self.order_id.company_id.security_lead)
+        )
```

Comment:
`res["date_planned"]` should already be the delivery date without the security lead, right?

Reply:
In 14.0 the computations are (done by the call to `super` right before):
```
date_deadline = commitment_date or (date_order + customer_lead)
date_planned = date_deadline - security_lead
```
I could even avoid these lines to compute the `date_planned`, it's already done by `super()`... just wanted to be clear about the intent.